src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String DELIMITER = ",";
+    private static final String LS = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] totals = readFromFile(fromFileName);
+        String report = createReport(totals[0], totals[1]);
+        writeToFile(toFileName, report);
+    }
+
+    private int[] readFromFile(String fromFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.isBlank()) {
+                    continue;
+                }
+                String[] parts = line.split(DELIMITER);
+                if (parts.length != 2) {
+                    continue;
+                }
+                String operation = parts[0].trim();
+                int amount = Integer.parseInt(parts[1].trim());
+
+                if (SUPPLY.equals(operation)) {
+                    supplyTotal += amount;
+                } else if (BUY.equals(operation)) {
+                    buyTotal += amount;
+                }
```

Unknown operations are currently ignored (only `supply` and `buy` are handled, other values are silently dropped). The requirements say the operation types are `supply` or `buy` and unexpected operations should not be tolerated. Quote: **"Operation types are supply or buy."**

If `operation` is not `supply` or `buy`, throw a RuntimeException that includes the invalid operation value, the file name, and the line number. This enforces strict input validation and prevents silent corruption when input contains typos or unexpected tokens.